libs/ml/include/ml/core/node.hpp

```diff
@@ -141,6 +143,105 @@ class Node
     return static_cast<bool>(cached_output_status_ == CachedOutputState::VALID_CACHE);
   }
 
+  virtual uint64_t ForwardPassChargeCost()
+  {
+    static constexpr uint64_t my_default_cost = 100;  // TODO(VH): Calculate/measure me
```

Comment:
even if we only use this in one function I would suggest moving this static constexpr down to the private member variables so it's easy to see what are the initialised/hard-coded parameters all in one place

Reply:
reworked: now any non-placeholder inputless (leaf) Node returns the same charge as a Placeholder one, and the estiamtion depends on output shape.